src/lib/models/movie.ts

```diff
@@ -0,0 +1,128 @@
+import fetch from 'node-fetch';
+import { GENRES } from '../genreMap';
+import {
+  GenreID,
+  Movie,
+  MovieDetailsFromAPI,
+  MovieFromAPI,
+  MoviesFromAPI,
+} from '../types';
+const API_KEY = process.env.API_KEY;
+
+export async function getPopularMovies(): Promise<Movie[]> {
+  const response = await fetch(
+    `https://api.themoviedb.org/3/movie/popular?api_key=${API_KEY}&language=en-US`
+  );
+  const movies: MoviesFromAPI = await response.json();
+  const formattedMovies = movies.results.map((movie: MovieFromAPI) => {
+    return {
+      title: movie.title,
+      rating: movie.vote_average,
+      overview: movie.overview,
+      image: `https://image.tmdb.org/t/p/w500${movie.poster_path}`,
+      id: movie.id,
+      genres: movie.genre_ids.map((id: GenreID) =>
+        GENRES[id] ? GENRES[id] : ''
+      ),
+    };
```

Comment:
You reapeat this snippet multiple times. Maybe consider creating a formatMovies function.